cluster-state-service/canary/pkg/tests/list_instances.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2016-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/blox/blox/cluster-state-service/canary/pkg/tests/util"
+	"github.com/blox/blox/cluster-state-service/canary/pkg/wrappers"
+	"github.com/pkg/errors"
+)
+
+const (
+	listInstancesClusterNamePrefix = canaryClusterNamePrefix + "ListInstances_"
+)
+
+func ListInstances(ecsWrapper wrappers.ECSWrapper, ec2Wrapper wrappers.EC2Wrapper, cssWrapper wrappers.CSSWrapper) error {
+	clusterName := listInstancesClusterNamePrefix + util.TimeString()
+
+	err := ecsWrapper.CreateCluster(&clusterName)
+	if err != nil {
+		return err
+	}
+	fmt.Println("List instances - created cluster")
+
+	numInstances := int64(1)
+	ec2InstanceIDs, err := ec2Wrapper.LaunchInstances(&numInstances, &clusterName)
+	if err != nil {
+		return err
+	}
+	fmt.Println("List instances - launched EC2 instance")
+
+	instanceARN, err := util.ValidateECSContainerInstanceAndGetInstanceARN(ecsWrapper, clusterName)
+	if err != nil {
+		return err
+	}
+	fmt.Println("List instances - instance registered with ECS")
+
+	err = validateListInstancesCall(cssWrapper, clusterName, instanceARN)
+	if err != nil {
+		return err
+	}
+	fmt.Println("List instances - instance in CSS ListInstances response")
+
+	err = util.DeleteCluster(ecsWrapper, clusterName)
+	if err != nil {
+		return err
+	}
+
+	err = util.TerminateInstances(ec2Wrapper, ec2InstanceIDs)
+	if err != nil {
+		return err
+	}
+	fmt.Println("List instances - ListInstances test resources cleaned up")
+
+	return nil
+}
+
+func validateListInstancesCall(cssWrapper wrappers.CSSWrapper, clusterName string, instanceARN string) error {
+	// Takes some time for CSS to get the instance.
+	// Retry list call once every 10 seconds for 2 minutes.
+	found := false
+	for i := 0; i < 12; i++ {
+		time.Sleep(10 * time.Second)
+		instances, err := cssWrapper.ListInstances(&clusterName)
+		if err != nil {
+			return err
+		}
+		for _, ins := range instances {
+			if *ins.ContainerInstanceARN == instanceARN {
```

Comment:
we should have more than one instance to make sure it doesn't get overwritten or anything weird in css, ie cluster can contain more than one instance and they will be returned correctly